# Exploration et Nettoyage de Données Sell4All

Ce notebook contient l'exploration initiale et le nettoyage des données clients pour le projet d'intelligence artificielle de Sell4All.

In [4]:
import matplotlib.pyplot as plt
import pandas as pd

## 1 Lecture des Données et Aperçu

Nous commençons par charger le fichier de données `dataset-sell4all.csv` et afficher ses 5 premières lignes.

In [5]:
df = pd.read_csv('dataset-sell4all.csv')

#affichage des 5 premières lignes
df.head()

,Name,Phone Number,Email,Address,Country,Postal code,Last date of connection,Last time of connection,Age,Gender,Customer spendings
0,Aaron Cote,966-7625,elit@hotmail.org,699-5837 Risus Street,Norway,4126,5-Apr-21,4:39,71,Man,356
1,Angelica Lawson,232-3051,diam.proin@google.org,481-8428 Magna. Street,Pakistan,518885,"oct. 10, 2021",0:36,37,Women,173
2,Louis Gilbert,1-997-733-0134,lorem.fringilla@hotmail.org,Ap #192-2082 Enim. Ave,Colombia,575444,2-Jul-22,3:37,24,Women,105
3,Basia Finley,1-987-322-7148,tristique.aliquet@icloud.co,608-2732 Nec Rd.,South Africa,5973-5765,19-Feb-22,2:34,37,Women,28
4,Rhona Sears,387-7682,iaculis.odio.nam@protonmail.org,8763 In Rd.,France,37476,31-Mar-22,6:28,42,Women,13


## 2. Résumé Technique des Données

Pour obtenir un résumé technique complet de notre jeu de données (nombre de lignes, colonnes, types de données, valeurs manquantes), nous utilisons la méthode `info()` de Pandas.

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 505 entries, 0 to 504
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   Name                     505 non-null    str  
 1   Phone Number             505 non-null    str  
 2   Email                    505 non-null    str  
 3   Address                  505 non-null    str  
 4   Country                  505 non-null    str  
 5   Postal code              505 non-null    str  
 6   Last date of connection  505 non-null    str  
 7   Last time of connection  505 non-null    str  
 8   Age                      505 non-null    int64
 9   Gender                   505 non-null    str  
 10  Customer spendings       505 non-null    int64
dtypes: int64(2), str(9)
memory usage: 43.5 KB


### Explication du Résumé Technique

1. **Combien y a-t-il d'entrées (lignes)?**
   * Le jeu de données contient **505 entrées** (lignes), indexées de `0` à `504`

2. **Qu'est-ce que « non-nul » ?**
   * Une valeur **non-nulle** signifie que la cellule correspondante contient une donnée valide et n'est pas vide 
   * Ici, toutes nos colonnes affichent `506 non-null`, ce qui indique qu'**il n'y a aucune valeur manquante** dans l'ensemble du fichier original

3. **Quels types de données sont présents dans l'ensemble de données et que signifient-ils ?**
   * Deux types de données principaux sont détectés par Pandas :
     * **`int64` (Entiers de 64 bits)** : Concerne les valeurs numériques sans décimales
     Présent dans 2 colonnes :
       * `Age` : L'âge des clients
       * `Customer spendings` : Les dépenses des clients en euros
     * **`str` (Texte / Chaînes de caractères)** : représente des données textuelles
     Présent dans 9 colonnes :
       * `Name`, `Phone Number`, `Email`, `Address`, `Country`, `Last date of connection`, `Last time of connection`, `Gender`
       * `Postal code` est également détecté comme `str` car de nombreux pays utilisent des codes postaux contenant des lettres 